### PR TITLE
GH-12224: refactor: Remove checkout dependencies from saved cart

### LINKED ISSUE #12224: Dependency cleanup -> [cart/saved-cart] -> ClearCheckoutService
Fire an event in the saved cart feature + listen for this event in the checkout lib and call the exact same service as before.

```
feature-libs/cart/saved-cart/components/details/saved-cart-details-action/saved-cart-details-action.component.ts:12:3 - error TS2305: Module '"@spartacus/core"' has no exported member 'ClearCheckoutService'.
12   ClearCheckoutService,

feature-libs/cart/saved-cart/components/list/saved-cart-list.component.ts:10:3 - error TS2305: Module '"@spartacus/core"' has no exported member 'ClearCheckoutService'.
10   ClearCheckoutService,

feature-libs/cart/saved-cart/components/saved-cart-form-dialog/saved-cart-form-dialog.component.ts:19:3 - error TS2305: Module '"@spartacus/core"' has no exported member 'ClearCheckoutService'.
19   ClearCheckoutService,
```

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-12224

### CODE REVIEW

File: feature-libs/cart/saved-cart/components/details/saved-cart-details-action/saved-cart-details-action.component.ts [hunk 1]
```diff
@@ -31,51 +26,13 @@ export class SavedCartDetailsActionComponent implements OnInit, OnDestroy {
...
-  constructor(
-    savedCartDetailsService: SavedCartDetailsService,
-    savedCartService: SavedCartFacade,
-    routingService: RoutingService,
-    globalMessageService: GlobalMessageService,
-    savedCartFormLaunchDialogService: SavedCartFormLaunchDialogService,
-    vcr: ViewContainerRef,
-    clearCheckoutService: ClearCheckoutService
-  );
-
```
Comment (plabadie) [inline feature-libs/cart/saved-cart/components/details/saved-cart-details-action/saved-cart-details-action.component.ts:70]: Breaking changes for this constructor will be handled in GH-12567

File: feature-libs/checkout/core/events/checkout-event.listener.ts [hunk 2]
```diff
@@ -42,6 +52,24 @@ export class CheckoutEventListener implements OnDestroy {
...
 
+  protected onSaveCartSuccess() {
+    this.subscription.add(
+      this.eventService.get(SaveCartSuccessEvent).subscribe((_event) => {
+        this.clearCheckoutFacade.resetCheckoutProcesses();
+      })
+    );
+  }
+
+  protected onRestoreSavedCartSuccess() {
```
Comment (WeizhengSap) [inline feature-libs/checkout/core/events/checkout-event.listener.ts]: maybe these 2 functions can be merged into one just like `onUserAddressChange`. Listen to `SavedCartEvent`, and only if `SaveCartSuccessEvent` and `RestoreSavedCartSuccessEvent`, then resetCheckoutProcesses.

Comment (plabadie) [reply feature-libs/checkout/core/events/checkout-event.listener.ts:63]: I suggest we align the address book event to be consistent with the ones ini this PR.  Followup ticket GH-12630

Comment (plabadie) [reply feature-libs/checkout/core/events/checkout-event.listener.ts]: Fixed

**Review by WeizhengSap — APPROVED** [review]

**Pull request merged**